balboa-agent/src/main/java/com/socrata/balboa/agent/MetricConsumer.java

```diff
@@ -147,7 +157,6 @@ public Record(String entityId, String name, Number value, long timestamp, Metric
             while (true) {
                 Map<String, String> record = grovel(stream);
                 if (record == null) {
-                    log.error("Unable to process records from {}", filePath);
                     break;
```

Comment:
I know you didn't name that function, but `grovel` is a terrible name. `readRecord` seems to succinctly summarize what it is doing?

What is that big while loop with a label even doing in there? I can't see it's purpose.

It seems like:
0xfe = End of Record
0xff = Beginning of Record

??